tests/Microsoft.Identity.Test.Integration.netfx/HeadlessTests/CaeAppTests.cs

```diff
@@ -0,0 +1,93 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Threading.Tasks;
+using Microsoft.Identity.Client;
+using Microsoft.Identity.Test.LabInfrastructure;
+using Microsoft.Identity.Test.Unit;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+namespace Microsoft.Identity.Test.Integration.NetFx.HeadlessTests
+{
+    [TestClass]
+    public class CaeAppTests
+    {
+        private string _confidentialClientSecret;
+        private readonly string _caeAppClientId = LabApiConstants.LabCaeConfidentialClientId;
+        private readonly string _tenantId = "72f988bf-86f1-41af-91ab-2d7cd011db47";
+        private readonly KeyVaultSecretsProvider _keyVaultProvider = new(KeyVaultInstance.MsalTeam);
+        private readonly HttpClient _httpClient = new();
+        private readonly TimeSpan _delayTimeout = TimeSpan.FromMinutes(5);
+        private readonly string[] _graphAppScope = ["https://graph.microsoft.com/.default"];
+
+        [TestInitialize]
+        public async Task TestInitialize()
+        {
+            if (string.IsNullOrEmpty(_confidentialClientSecret))
```

Comment:
can this ever be empty?

Reply:
Has to be set. Kept it as an instance variable so it's easier to set for manual testing.